app/src/main/java/io/homeassistant/companion/android/settings/shortcuts/ManageShortcutsSettingsFragment.kt

```diff
@@ -0,0 +1,311 @@
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported)
+                            pinPreference?.isVisible = true
+                    } else {
+                        addUpdatePreference?.isEnabled = false
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported)
+                            pinPreference?.isVisible = false
+                    }
+                    return@setOnPreferenceChangeListener true
+                }
+            }
+
+            findPreference<EditTextPreference>(SHORTCUT_PREFIX + i + PATH_SUFFIX)?.let {
+                it.setOnPreferenceChangeListener { _, newValue ->
+                    shortcutPath = newValue.toString()
+                    if (!shortcutLabel.isNullOrEmpty() && !shortcutDesc.isNullOrEmpty() && !shortcutPath.isNullOrEmpty()) {
+                        addUpdatePreference?.isEnabled = true
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported)
+                            pinPreference?.isVisible = true
+                    } else {
+                        addUpdatePreference?.isEnabled = false
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported)
+                            pinPreference?.isVisible = false
+                    }
+                    return@setOnPreferenceChangeListener true
+                }
+            }
+
+            addUpdatePreference?.setOnPreferenceClickListener {
+                Log.d(TAG, "Creating shortcut #: $i")
+                if (!shortcutLabel.isNullOrEmpty() && !shortcutDesc.isNullOrEmpty() && !shortcutPath.isNullOrEmpty()) {
+                    val shortcut = createShortcut(SHORTCUT_PREFIX + i, shortcutLabel!!, shortcutDesc!!, shortcutPath!!)
+                    shortcutManager!!.addDynamicShortcuts(listOf(shortcut))
+                }
+                dynamicShortcuts = shortcutManager.dynamicShortcuts
+                it.title = getString(R.string.update_shortcut)
+                deletePreference?.isVisible = true
+                return@setOnPreferenceClickListener true
+            }
+
+            deletePreference?.setOnPreferenceClickListener {
+                Log.d(TAG, "Attempting to delete shortcut #: $i")
+                shortcutManager.removeDynamicShortcuts(listOf(SHORTCUT_PREFIX + i))
+                dynamicShortcuts = shortcutManager.dynamicShortcuts
+                addUpdatePreference?.title = getString(R.string.add_shortcut)
+                it.isVisible = false
+                return@setOnPreferenceClickListener true
+            }
+
+            pinPreference?.setOnPreferenceClickListener {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported) {
```

Comment:
This condition is used really often. Maybe it is possible to refactor it as method.

Reply:
So the first case this condition is used is for the 5 hard coded dynamic shortcuts to show that they can be pinned (if supported) The next time this condition is used is to hide the pinned category if the users device does not support it. I didnt see a good way to refactor this since the underlying code is different

Reply:
actually there is more than 2 instances but has a different purpose, will need to see if i can refactor some of them

Reply:
Something like this:

Instead of 
```kotlin
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported)
```

something like this one:
```kotlin
if (pinShortcutsSupported())
```

Reply:
Now that we are removing the ability to pin the dynamic shortcuts there is only one instance of this check :) So doesnt make sense but a lot less code now